lib/nats/io/msg.rb

```diff
@@ -40,6 +42,22 @@ def respond(data = "")
       end
     end
 
+    def respond_with_error(error)
```

Comment:
Notice that the python client distinguishes between `Msg` and `Request` types to be able to extend different behaviors, for example here `respond_with_error`. Do you think we could change this into being part of a `Request` type instead of adding this to the core `Msg` type? e.g. https://github.com/nats-io/nats.py/blob/60f58f659dc5312a0bfb3df4099b7345d0c98af8/nats/micro/request.py#L67

This way in case the service Request adr spec is expanded then do not need to change the core Msg type.